report.go

```diff
@@ -14,14 +14,15 @@ import (
 	"text/tabwriter"
 )
 
-// report generates a failure report.
-func report(msg string, a ...interface{}) string {
+// report generates a failure report for the given error, optionally including
+// the in the output the given comment
+func report(err error, c Comment) string {
 	var buf bytes.Buffer
 	buf.WriteString("\n")
-	if len(a) > 0 {
-		fmt.Fprintln(&buf, a...)
+	if comment := c.String(); comment != "" {
+		fmt.Fprintln(&buf, comment)
 	}
-	fmt.Fprintln(&buf, strings.TrimSuffix(msg, "\n"))
+	fmt.Fprintln(&buf, strings.TrimSuffix(err.Error(), "\n"))
```

Comment:
I don't understand this (perhaps relevant to mhilton's remark earlier). Why would an error have a newline at the end of it?